core/platform-api-client-gwt/che-core-client-gwt-git/src/main/java/org/eclipse/che/api/git/gwt/client/GitServiceClient.java

```diff
@@ -357,6 +357,26 @@ void push(String workspaceId,
               String remote, boolean force,
               AsyncRequestCallback<PushResponse> callback);
 
+
+    /**
+     * Push changes from local repository to remote one (sends request over WebSocket).
+     *
+     * @param wsId
+     *         id of current workspace
+     * @param project
+     *         project
+     * @param refSpec
+     *         list of refspec to push
+     * @param remote
+     *         remote repository name or url
+     * @param force
+     *         push refuses to update a remote ref that is not an ancestor of the local ref used to overwrite it. If <code>true</code>
+     *         disables the check. This can cause the remote repository to lose commits
+     */
+    Promise<PushResponse> push(String wsId,
```

Comment:
Also it should be fixed over other issue, not related to this, here I only rewrite methods with the promises without changing in the common signature.